test/fw/ptl/utils/plugins/ptl_test_db.py

```diff
@@ -1615,15 +1615,19 @@ def __init__(self, dbtype, dbpath, dbaccess):
         self.res_data = PTLJsonData(command=self.__cmd)
 
     def __write_test_data(self, data):
-        jdata = None
         if _TESTRESULT_TN not in self.__dbobj.keys():
-            self.__dbobj[_TESTRESULT_TN] = open(self.dbpath, 'w+')
+            self.__dbobj[_TESTRESULT_TN] = self.dbpath
+            jdata = None
+            dbobj = self.__dbobj[_TESTRESULT_TN]
         else:
-            self.__dbobj[_TESTRESULT_TN].seek(0)
-            jdata = json.load(self.__dbobj[_TESTRESULT_TN])
+            dbobj = self.__dbobj[_TESTRESULT_TN]
+            with open(dbobj, 'r') as fd:
+                jdata = json.load(fd)
         jsondata = self.res_data.get_json(data=data, prev_data=jdata)
-        self.__dbobj[_TESTRESULT_TN].seek(0)
-        json.dump(jsondata, self.__dbobj[_TESTRESULT_TN], indent=2)
+        with open(dbobj, 'w') as fd:
+            json.dump(jsondata, fd, indent=2)
+            fd.flush()
```

Comment:
do you need the flush()? dump() also does a write(), so the next write() should just get buffered behind that, and all flushed when with block ends, right?

Reply:
I think you are correct, I will remove the flush().